sklearn/utils/_mocking.py

```diff
@@ -135,7 +196,7 @@ def fit(self, X, y, **fit_params):
 
         return self
 
-    def predict(self, X):
+    def predict(self, X, y=None):
```

Comment:
Having `y` in `predict`, `predict_proba` and `decision_function` is a little strange. What do you think about and not having `y=None` in the method signature?

```py
X, _ = self._check_X_y(X)
```

Reply:
yes my bad